apps/cowswap-frontend/src/modules/permit/types.ts

```diff
@@ -0,0 +1,42 @@
+import { latest } from '@cowprotocol/app-data'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Web3Provider } from '@ethersproject/providers'
+import { Token } from '@uniswap/sdk-core'
+
+export type SupportedPermitInfo = {
+  type: 'dai' | 'permit'
+  gasLimit: number
+}
+type UnsupportedPermitInfo = false
+export type PermitInfo = SupportedPermitInfo | UnsupportedPermitInfo
+
+export type PermittableTokens = Record<SupportedChainId, Record<string, PermitInfo>>
+
+export type IsTokenPermittableResult = PermitInfo | undefined
+
+export type AddPermitTokenParams = {
+  chainId: SupportedChainId
+  tokenAddress: string
+  permitInfo: PermitInfo
+}
+
+export type PermitHookParams = {
+  inputToken: Token
+  chainId: SupportedChainId
+  permitInfo: SupportedPermitInfo
+  provider: Web3Provider
+  account?: string
+}
+
+export type PermitHookData = latest.CoWHook | undefined
```

Comment:
I would not define here permit hook data as `export type PermitHookData = latest.CoWHook | undefined` and only as `export type PermitHookData = latest.CoWHook`

This way, you can signal when the data can be undefined or not. i.e. `export function usePermitHookData(params?: PermitHookParams): PermitHookData | undefined`

note that you added `| undefined` in the hook too, so now the hook returns technically `PermitHookData | undefined | undefined` :P